projects/plugins/jetpack/_inc/blogging-prompts.php

```diff
@@ -47,6 +47,35 @@ function jetpack_setup_blogging_prompt_response( $post_id ) {
 
 add_action( 'wp_insert_post', 'jetpack_setup_blogging_prompt_response' );
 
+/**
+ * When a published posts answers a blogging prompt, store the prompt id in the post meta.
+ *
+ * @param string  $new_status New post status.
+ * @param string  $old_status Old post status.
+ * @param WP_Post $post       Post object.
+ */
+function jetpack_mark_if_post_answers_blogging_prompt( $new_status, $old_status, $post ) {
+	// Make sure we are publishing a post, and it's not already published.
+	if ( 'post' !== $post->post_type || 'publish' !== $new_status || 'publish' === $old_status ) {
+		return;
+	}
+
+	$blocks = parse_blocks( $post->post_content );
+	foreach ( $blocks as $block ) {
+		if ( 'jetpack/blogging-prompt' === $block['blockName'] ) {
+			$prompt_id      = isset( $block['attrs']['promptId'] ) ? $block['attrs']['promptId'] : null;
+			$has_prompt_tag = has_tag( 'dailyprompt', $post ) || ( $prompt_id && has_tag( "dailyprompt-{$prompt_id}", $post ) );
```

Comment:
When testing with a new post (on wordpress.com), I found that has_tag('dailyprompt') would return false here even though the tag had already been applied to the post.

When I save the post as a draft and _then_ publish, I found that the post already had the tag applied.

I wonder if we need to refresh the post from the database to get the tags on a newly created post?

Reply:
@creativecoder it didn't actually add the metadata for me when I published a post. I was testing on wordpress.com, with this diff applied to my sandbox.

If I save the post as draft first, and then publish, it would apply the metadata correctly.

Reply:
Thanks for testing, @roo2 .

I can see the same problem testing on a Simple site. I think what's happening is that the tags aren't set until the tag panel is expanded, or until the post is saved as a draft. I'll look into a solution.